### DISCUSSION TITLE
File Upload: Unable to retrieve the file_size for file at location

### DISCUSSION THREAD

**Discussion by quocnguyen2001** [livewire/livewire#5734] [7. Bugs (archived)]
Hello everyone,

I have a weird issue. When the file name is "farming-agriculture-field-with-copy-space-blue-sky-overgrown-grass-fenced-isolated-farming-area-farm-with-lush-trees-yellow-green-grass-looks-beautiful-rural-countryside-nature_590464-74263.jpg", I got an issue "Unable to retrieve the file_size for file at location"

<img width="1440" alt="image" src="https://github.com/livewire/livewire/assets/95564912/3095ceac-91ce-4072-a6d9-42bb698e8478">

I tested on Livewire component, also Filament. Checked on Network tap, It sent uploaded file api, but null path return.
Any image files, just rename the same above.

You can test it on https://demo.filamentphp.com

Any thougth about it?

**Comment by tomphilpotts** [comment]
Have you run `php artisan storage:link` ?

**Reply by dinhquochan** [reply]
Of course, only this filename is unable to upload. I have the same issue but with another name.

**Comment by dinhquochan** [comment]
I have the same issue but with another name.

**Comment by njoguamos** [comment]
I have the same issue
```bash
Unable to retrieve the file_size for file at location: temp-uploads/gr65G6lkmyrbcNAMrhTQXxpqpMUonz-metaMzU0MjUxNTgwXzY1MzMwMzY2MzUxMDY1MF81NDA0NTg1NDExNjg4MjI0NjI3X24uanBn-.jpg. 
```

Laravel v10.13.5
Livewire v2.12.3

**Comment by Kenny201** [comment]
There was the same problem, was it possible to find a solution to this problem?

**Comment by oldevill14** [comment]
vendor\livewire\livewire\src\TemporaryUploadedFile.php

```
public static function generateHashNameWithOriginalNameEmbedded($file)
    {
        $string = str()->substr($file->getClientOriginalName(),0,128);
        $genNewOriginalName = preg_replace('/\s+/', '', $string);
        $hash = str()->random(30);
        $meta = str('-meta'.base64_encode($genNewOriginalName).'-')->replace('/', '_');
        $extension = '.'.$file->guessExtension();
        return $hash.$meta.$extension;
    }
```

**Comment by BVFI** [comment]
I see a couple of problems, the most obvious one is the length of the file name. Any uploaded file that you plan on keeping should always be renamed by the app and shouldnt be too long, its good practice. Always rename user uploaded files. 
This error can be thrown by unlink() & $this->validate() as well. If unlink()/validate() is looking to delete a file in livewire-tmp and the file doesnt exist there, it will throw this exact same error for some reason, which doesnt indicate that the file doesnt exists, just that its "file_size" doesnt. Well ofc the "file_size" wont exist if the file doesnt even exist, but sometimes be like that. Could be async issues as well (If the file gets deleted too early and then gets checked) I manually handle each step of the upload process and validate it against a couple of requirements and [safety measures](https://cheatsheetseries.owasp.org/cheatsheets/File_Upload_Cheat_Sheet.html)
The way I do it:
1. User uploads a file that goes through a couple of Alpine.JS checks before going to the server side
2. File gets stored into a public $uploadedFiles[]. 
3. On updatedUploadedFiles() I validate each file and store it in a new $files[] (The ones that dont pass validation, I unlink from the livewire-tmp) that I can use to display to the users. The on save() I store my $files to another dir and remove all of em in the tmp dir (This way you also avoid the issue with uploading multiple files and them resetting each time you upload a new batch of multiple files, so you basically NEED to have a $files array, If you want to let the users append uploaded files. Basically, this manual process lets me remove the files one by one and I wait for them to finish, before moving on. 

And the third reason I could think of that this could happen is because of file perms on the server. Use a debugger to figure out where exactly its triggered.

**Comment by D076** [comment]
livewire v3.5.4

Problem with Livewire\Features\SupportFileUploads\TemporaryUploadedFile generateHashNameWithOriginalNameEmbedded function.

If the original file name exceeds approximately 90 characters, a very long hash is generated, which prevents the file from being saved to the temporary directory.

```
    namespace Livewire\Features\SupportFileUploads;
    ...
    class TemporaryUploadedFile extends UploadedFile
    ...

    public static function generateHashNameWithOriginalNameEmbedded($file)
    {
        $hash = str()->random(30);
        $meta = str('-meta'.base64_encode($file->getClientOriginalName()).'-')->replace('/', '_');
        $extension = '.'.$file->getClientOriginalExtension();

        return $hash.$meta.$extension;
    }
```

The solution is add string limit to $file->getClientOriginalName() in your custom class and use AliasLoader to change alias.

Upd. Same in v3.5.6

**Reply by shazimali** [reply]
getting same problem, can you please explain how i can achieve above solution ?

**Reply by oldevill14** [reply]
> getting same problem, can you please explain how i can achieve above solution ?


TemporaryUploadedFile The characters that are too long can be used to create a hast name on the server.

There, the idea is to reduce the size of the characters to be short enough to be used to create a hast name
Or if you can increase the memory usage on the server, you can add this part on the server instead.

**Comment by sineld** [comment]
Any chance to merge this to main? @calebporzio

**Comment by fabiomlferreira** [comment]
This issue must be fixed

**Comment by pablozagni** [comment]
I solved it runing `chmod 777 -R storage`

**Reply by ahinkle** [reply]
**Don't advise doing this**. 777 gives full read/write/execute permissions to every user that can access the machine.

**Comment by alisalehi1380** [comment]
![image](https://github.com/user-attachments/assets/143e9a76-86f9-42a0-a84e-ca824c8a80b2)

I had written `nullable` and `sometimes` incorrectly and got this error.

I removed `sometimes` and it was fixed. ✅

**Comment by jhjvandenbroek** [comment]
+1 

Can't seem to solve this without editing the livewire files

**Comment by Naoray** [comment]
This is solved with https://github.com/livewire/livewire/pull/9172 ... just need someone to merge it

**Comment by learncodes12** [comment]
Hello I am getting same error on laravel filament when i am trying to upload the file with long length filename. What could be the solution for this ?
![image](https://github.com/user-attachments/assets/5e9d6d82-17eb-4722-bdc8-3629f94ea4d0)

**Reply by Fun-cly** [reply]
did you set two rules?
like on config/livewire and on modelresource?

**Comment by DarkcoderSe** [comment]
Is it still solved or not?

**Comment by anas322** [comment]
i solved this probelm by make a middleware work around, 

```
<?php

namespace App\Http\Middleware;

use Closure;
use Illuminate\Http\Request;
use Illuminate\Http\UploadedFile;
use Illuminate\Support\Facades\Log;
use Symfony\Component\HttpFoundation\Response;

/**
 * Middleware to truncate long filenames for Livewire file uploads
 *
 * Livewire's TemporaryUploadedFile::generateHashNameWithOriginalNameEmbedded()
 * base64-encodes the original filename, which can exceed Windows' 255 char path limit.
 * This middleware renames uploaded files with long names before Livewire processes them.
 */
class TruncateLivewireFilename
{
    /**
     * Maximum filename length (excluding extension) to avoid path length issues.
     */
    private const MAX_FILENAME_LENGTH = 50;

    /**
     * Handle an incoming request.
     *
     * @param  \Closure(\Illuminate\Http\Request): (\Symfony\Component\HttpFoundation\Response)  $next
     */
    public function handle(Request $request, Closure $next): Response
    {
        $path = $request->path();

        // Only process Livewire file upload requests
        if (! str_contains($path, 'livewire/upload-file')) {

            return $next($request);
        }


        $files = $request->allFiles();

        foreach ($files as $key => $file) {
            $this->processFiles($key, $file);
        }

        return $next($request);
    }

    /**
     * Process files recursively (handles nested arrays from Livewire).
     */
    private function processFiles(string $key, mixed $file): void
    {
        if (is_array($file)) {
            foreach ($file as $subKey => $subFile) {
                $this->processFiles($key . '.' . $subKey, $subFile);
            }

            return;
        }

        if ($file instanceof UploadedFile) {
            $this->truncateFilenameIfNeeded($file);
        } else {
        }
    }

    /**
     * Truncate the filename if it exceeds the max length.
     */
    private function truncateFilenameIfNeeded(UploadedFile $file): void
    {
        $originalName = $file->getClientOriginalName();
        $extension = pathinfo($originalName, PATHINFO_EXTENSION);
        $basename = pathinfo($originalName, PATHINFO_FILENAME);


        // Only truncate if the basename exceeds max length
        if (mb_strlen($basename) <= self::MAX_FILENAME_LENGTH) {

            return;
        }

        // Truncate basename and preserve extension
        $truncatedBasename = mb_substr($basename, 0, self::MAX_FILENAME_LENGTH);
        $newFilename = $extension
            ? "{$truncatedBasename}.{$extension}"
            : $truncatedBasename;


        // Use reflection to update the private originalName property on Symfony's UploadedFile
        // The property is on the parent class (Symfony\Component\HttpFoundation\File\UploadedFile)
        $symfonyClass = \Symfony\Component\HttpFoundation\File\UploadedFile::class;
        $reflection = new \ReflectionClass($symfonyClass);

        // Get the originalName property from Symfony's UploadedFile
        if ($reflection->hasProperty('originalName')) {
            $property = $reflection->getProperty('originalName');
            $property->setAccessible(true);
            $property->setValue($file, $newFilename);
        } else {
            Log::warning('[TruncateLivewireFilename] Could not find originalName property on Symfony UploadedFile');

            // Log all properties for debugging
            $properties = $reflection->getProperties();
            foreach ($properties as $prop) {
            }
        }

        // Verify the change
        $verifyName = $file->getClientOriginalName();
    }
}

```

and register in the app bootstrap 

<img width="1198" height="692" alt="image" src="https://github.com/user-attachments/assets/239b3bda-1470-4a37-b11e-bd9b57516067" />
